enterprise/server/remote_execution/workspace/workspace_test.go

```diff
@@ -139,3 +139,93 @@ func TestWorkspaceCleanup_PreserveWorkspace_PreservesAllFilesExceptOutputs(t *te
 		"expected all KEEPME filePaths (and no others) in the workspace after cleanup",
 	)
 }
+
+func TestCleanInputsIfNecessary_CleanNone(t *testing.T) {
+	filePaths := []string{
+		"some_input_directory/foo.framework/KEEPME",
+		"some/nested/input/directory/KEEPME.h",
+		"some_input_file_KEEPME",
+		"some/nested/input/file/KEEPME",
+		"KEEPME",
+		"foo/KEEPME",
+		"foo/bar/KEEPME",
+	}
+	ws := newWorkspace(t, &workspace.Opts{Preserve: true})
+	writeEmptyFiles(t, ws, filePaths)
+
+	for _, file := range filePaths {
+		ws.Inputs[file] = &repb.Digest{}
+	}
+
+	keep := map[string]*repb.Digest{
+		"KEEPME":         &repb.Digest{},
+		"foo/KEEPME":     &repb.Digest{},
+		"foo/bar/KEEPME": &repb.Digest{}}
+
+	ws.CleanInputsIfNecessary(keep)
```

Comment:
nit: assert that there's no error:

```
err := ws.CleanInputsIfNecessary(keey)

require.NoError(t, err)
```

(same in the test below)

Reply:
Done